anti-diff/src/Data/FingerTree/RootMeasured/Strict.hs

```diff
@@ -0,0 +1,251 @@
+  , RootMeasured (..)
+  , SuperMeasured
+    -- * Construction
+  , fromList
+  , (|>)
+    -- * Splitting
+  , LR (..)
+  , split
+  , splitl
+  , splitlr
+  , splitr
+    -- * Maps
+  , fmap'
+  , fmap''
+  ) where
+
+import           Data.Foldable
+import           Data.Group
+import           GHC.Generics (Generic)
+
+import           NoThunks.Class (NoThunks (..), noThunksInValues)
+
+import           Data.FingerTree.Strict (Measured)
+import qualified Data.FingerTree.Strict as FT
+
+{-------------------------------------------------------------------------------
+  Strict finger trees with root measures
+-------------------------------------------------------------------------------}
+
+-- | A @StrictFingerTree@ with elements of type @a@, an internal measure of type
+-- @vi@, and a root measure of type @vt@.
+data StrictFingerTree vt vi a = SFT {
+    tm       :: vt
+  , elements :: !(FT.StrictFingerTree vi a)
+  }
+  deriving (Show, Eq, Ord, Generic)
+
+instance Foldable (StrictFingerTree vt vi) where
+  foldMap f = foldMap f . elements
+
+instance NoThunks a => NoThunks (StrictFingerTree vt vi a) where
+  showTypeOf _ = "StrictFingerTree'"
+  wNoThunks ctxt = noThunksInValues ctxt . toList
+
+instance (Semigroup vt, Measured vi a)
+      => Semigroup (StrictFingerTree vt vi a) where
+  SFT tm1 xs1 <> SFT tm2 xs2 = SFT (tm1 <> tm2) (xs1 FT.>< xs2)
+
+instance (Monoid vt, Measured vi a) => Monoid (StrictFingerTree vt vi a) where
+  mempty = SFT mempty FT.empty
+
+{-------------------------------------------------------------------------------
+  Measuring
+-------------------------------------------------------------------------------}
+
+-- | All @StrictFingerTree@s are internally measured.
+instance Measured vi a => Measured vi (StrictFingerTree vt vi a) where
+  measure = FT.measure . elements
+
+-- | Re-iteration of @'Measured'@, but for root measures.
+--
+-- This re-iteration is necessary because we want to allow the root measure to
+-- be distinct from the internal measure. For example, we can not create both of
+-- these instances for distinct types @T@ and @T'@:
+--
+-- > @instance Measured T  a where -- ...@
+--
+-- > @instance Measured T' a where -- ...@
+--
+-- Furthermore, we want the root measure to be a @'Group'@ instead of a
+-- @'Monoid'@.
+class Monoid v => RootMeasured v a | a -> v where
+  measureRoot :: a -> v
+
+-- | All @StrictFingerTree@s are root measured.
+instance RootMeasured vt a => RootMeasured vt (StrictFingerTree vt vi a) where
+  measureRoot = tm
+
+-- | Conjunction of @RootMeasured@ and @Measured@ constraints.
+type SuperMeasured vt vi a = (RootMeasured vt a, Measured vi a)
+
+{-------------------------------------------------------------------------------
+  Construction
+-------------------------------------------------------------------------------}
+
+infixl 5 |>
+
+-- | /O(1)/. Add an element to the right end of a sequence.
+--
+-- Mnemonic: a triangle with the single element at the pointy end.
```

Comment:
I think these comments are taken directly from `Data.Sequence`. Are they still true? i.e. is this still in `O(1)`?